utils/src/main/java/com/indix/gocd/utils/store/S3ArtifactStore.java

```diff
@@ -187,4 +199,13 @@ public RevisionStatus getLatest(AmazonS3Client client, Artifact artifact) {
         }
         return null;
     }
+
+    public static AmazonS3 getS3client(GoEnvironment env) {
+        AmazonS3ClientBuilder amazonS3ClientBuilder = AmazonS3ClientBuilder.standard();
```

Comment:
The below in the code you referenced:

`new EnvironmentVariableCredentialsProvider()` - Present
`new SystemPropertiesCredentialsProvider()` - Present
`new ProfileCredentialsProvider()` - Present
`new InstanceProfileCredentialsProvider()` - Present
`new BasicAWSCredentialsProvider(accessKey, secretKey)` - Not needed. Not that the code gets the creds from env vars and then passes as params.
`new AnonAWSCredentialsProvider()` - Present

Also adds ECS support :)